### ISSUE TITLE
Try again to download a torrent file later

### ISSUE THREAD

**Issue by christianmerges** [cross-seed/cross-seed#492]
Hi,

could u please implement to try again downloading a torrent file, when it doesn't succeed? I get the following error multiple times in the log:

error: error downloading torrent at http://prowlarr:9696/13/download?apikey=[REDACTED]&link=T0pRNjFCVkw5RHBLN041UGJGYU5oekVVc05tZXhOUFYyOTRvN3hVeTZUb3hNVlkyNTJrYzhrWngwWXoxZWlHT09VME9COEJoalowTmdzRUpxMmpVZ2xyUnFBZC9OdW9CbytubFVZeUpwM289&file=XXXXXXXX: 500 Internal Server Error

I assume this is rate limiting by the tracker.

**Comment by stevietv** [comment]
I am getting a 500 on cross-seed for a 429 rate limit from Orpheus.

prowlarr log:
```2023-09-23 14:18:05.4|Trace|IndexerHttpClient|Res: HTTP/2.0 [GET] https://orpheus.network/ajax.php?action=download&id=(removed)&usetoken=1: 200.OK (154 bytes) (216 ms)
2023-09-23 14:18:05.4|Trace|IndexerHttpClient|Req: [GET] https://orpheus.network/ajax.php?action=download&id=(removed)
2023-09-23 14:18:05.4|Trace|ConfigService|Using default config value for 'proxyenabled' defaultValue:'False'
2023-09-23 14:18:05.8|Trace|IndexerHttpClient|Res: HTTP/2.0 [GET] https://orpheus.network/ajax.php?action=download&id=(removed) 200.OK (117 bytes) (379 ms)
2023-09-23 14:18:05.8|Trace|Orpheus|Invalid torrent file contents: {"status":"failure","error":"rate limiting hit on downloading","response":[],"info":{"source":"Orpheus","version":1}}
2023-09-23 14:18:05.8|Error|NewznabController|MonoTorrent.TorrentException: Invalid torrent file specified

[v1.9.1.3981] MonoTorrent.TorrentException: Invalid torrent file specified
 ---> MonoTorrent.BEncoding.BEncodingException: Invalid data found. Aborting
   at MonoTorrent.BEncoding.BEncodeDecoder.DecodeTorrent(RawReader reader)
   at MonoTorrent.Torrent.Load(Stream stream, String path)
   --- End of inner exception stack trace ---
   at MonoTorrent.Torrent.Load(Stream stream, String path)
   at MonoTorrent.Torrent.Load(Byte[] data)
   at NzbDrone.Core.Indexers.TorrentIndexerBase`1.ValidateDownloadData(Byte[] fileData) in ./Prowlarr.Core/Indexers/TorrentIndexerBase.cs:line 28
   at NzbDrone.Core.Indexers.Definitions.Orpheus.Download(Uri link) in ./Prowlarr.Core/Indexers/Definitions/Orpheus.cs:line 117
   at NzbDrone.Core.Download.DownloadService.DownloadReport(String link, Int32 indexerId, String source, String host, String title) in ./Prowlarr.Core/Download/DownloadService.cs:line 191
   at NzbDrone.Api.V1.Indexers.NewznabController.GetDownload(Int32 id, String link, String file) in ./Prowlarr.Api.V1/Indexers/NewznabController.cs:line 300

2023-09-23 14:18:05.8|Trace|Http|Res: 3527 [GET] /20/download?apikey=(removed)&link=cHVVYThncnpYMno4MDJEbGk3ZStLNys1NlExZ01ucE90ZnJNcFlremdQeUJmQWlkR1Fac0tQTm03VE9uanRiaGlJaGxBYjVpeXBQUHRLYUQ4WVhDUE8wdlU1dmJkUDgvVVRrQ0tGRFNyUlRUOVpUQk4zc09HOVgzQUpOcVg3aEM&file=Ash+-+Race+the+Night+%5B2023%5D+%5BAlbum%5D+%5BFLAC+Lossless%5D+%5BWEB%5D: 500.InternalServerError (663 ms)
   ```

cross-seed log
```
2023-09-23 12:17:59 [32minfo[39m: [torznab] Searching 8 indexers for Ash - Race the Night (2023) - WEB FLAC
2023-09-23 12:17:59 [36mverbose[39m: [torznab] http://prowlarr:9696/16/api?apikey=[REDACTED]&t=search&q=Ash+-+Race+the+Night+2023
2023-09-23 12:17:59 [36mverbose[39m: [torznab] http://prowlarr:9696/15/api?apikey=[REDACTED]&t=search&q=Ash+-+Race+the+Night+2023
2023-09-23 12:17:59 [36mverbose[39m: [torznab] http://prowlarr:9696/10/api?apikey=[REDACTED]&t=search&q=Ash+-+Race+the+Night+2023
2023-09-23 12:17:59 [36mverbose[39m: [torznab] http://prowlarr:9696/17/api?apikey=[REDACTED]&t=search&q=Ash+-+Race+the+Night+2023
2023-09-23 12:17:59 [36mverbose[39m: [torznab] http://prowlarr:9696/13/api?apikey=[REDACTED]&t=search&q=Ash+-+Race+the+Night+2023
2023-09-23 12:17:59 [36mverbose[39m: [torznab] http://prowlarr:9696/8/api?apikey=[REDACTED]&t=search&q=Ash+-+Race+the+Night+2023
2023-09-23 12:17:59 [36mverbose[39m: [torznab] http://prowlarr:9696/19/api?apikey=[REDACTED]&t=search&q=Ash+-+Race+the+Night+2023
2023-09-23 12:17:59 [36mverbose[39m: [torznab] http://prowlarr:9696/20/api?apikey=[REDACTED]&t=search&q=Ash+-+Race+the+Night+2023
2023-09-23 12:18:05 [36mverbose[39m: [decide] Ash - Race the Night (2023) - WEB FLAC - no match for TorrentSeeds (API) torrent Ash-Race_the_Night-WEB-2023-ENViED - its size does not match
2023-09-23 12:18:05 [36mverbose[39m: [decide] Ash - Race the Night (2023) - WEB FLAC - no match for Redacted torrent Ash - Race the Night [2023] [Album] [MP3 320] [WEB] - its size does not match
2023-09-23 12:18:05 [36mverbose[39m: [decide] Ash - Race the Night (2023) - WEB FLAC - no match for Redacted torrent Ash - Race the Night [2023] [Album] [MP3 V0 (VBR)] [WEB] - its size does not match
2023-09-23 12:18:05 [36mverbose[39m: [decide] Ash - Race the Night (2023) - WEB FLAC - no match for Redacted torrent Ash - Race the Night [2023] [Single] [FLAC Lossless] [WEB] - its size does not match
2023-09-23 12:18:05 [36mverbose[39m: [decide] Ash - Race the Night (2023) - WEB FLAC - no match for Orpheus torrent Ash - Race the Night [2023] [Album] [MP3 V0 (VBR)] [WEB] - its size does not match
2023-09-23 12:18:05 [36mverbose[39m: [decide] Ash - Race the Night (2023) - WEB FLAC - no match for Orpheus torrent Ash - Race the Night [2023] [Album] [MP3 320] [WEB] - its size does not match
2023-09-23 12:18:05 [31merror[39m: error downloading torrent at http://prowlarr:9696/20/download?apikey=[REDACTED]&link=cHVVYThncnpYMno4MDJEbGk3ZStLNys1NlExZ01ucE90ZnJNcFlremdQeUJmQWlkR1Fac0tQTm03VE9uanRiaGlJaGxBYjVpeXBQUHRLYUQ4WVhDUE8wdlU1dmJkUDgvVVRrQ0tGRFNyUlRUOVpUQk4zc09HOVgzQUpOcVg3aEM&file=Ash+-+Race+the+Night+%5B2023%5D+%5BAlbum%5D+%5BFLAC+Lossless%5D+%5BWEB%5D: 500 Internal Server Error
2023-09-23 12:18:05 [34mdebug[39m: response: <?xml version="1.0" encoding="UTF-8"?>
<error code="500" description="Invalid torrent file specified" />
```

it seems that prowlarr is indeed returning a 500, because this specific indexer doesn't return the 429 but a json page which gets grabbed as a 'torrent'

**Comment by christianmerges** [comment]
For me the reason was, that the tracker blocked the download rights. I figured it out by trying to download a torrent file manually. But anyway the point is, looking (searching) for duplicates is a time consuming process, so in case a cross-seed is found, but couldn't be downloaded for whatever reason, it would be great to try again later and not having to check all the torrents again after fixing the problem with the tracker.

**Comment by mmgoodnow** [comment]
dupe of #402
